src/Server/RestRouterHandlers/SearchHandler.cpp

```diff
@@ -0,0 +1,219 @@
+
+#include <re2/re2.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int BAD_REQUEST_PARAMETER;
+}
+
+std::map<String, std::map<String, String>> SearchHandler::search_schema
+    = {{"required", {{"query", "string"}}},
+       {"optional", {{"mode", "string"}, {"end_time", "string"}, {"start_time", "string"}, {"offset", "int"}, {"page_size", "int"}}}};
+
+String SearchHandler::execute(ReadBuffer & input, HTTPServerResponse & response, Int32 & http_status) const
+{
+    auto send_error = [&http_status, &response](const String && error_msg)
+    {
+        http_status = HTTPResponse::HTTP_BAD_REQUEST;
+        response.setStatusAndReason(HTTPResponse::HTTPStatus(http_status));
+        *response.send() << error_msg << std::endl;
+        return error_msg;
+    };
+
+    String data = "{}";
+    auto size = getContentLength();
+    if (size > 0)
+    {
+        data.resize(size);
+        input.readStrict(data.data(), size);
+    }
+    Poco::JSON::Parser parser;
+    auto payload = parser.parse(data).extract<Poco::JSON::Object::Ptr>();
+
+    String error;
+    if (!validatePayload(payload, error))
+        return send_error(jsonErrorResponse(error, ErrorCodes::BAD_REQUEST_PARAMETER));
+
+    const auto & query = getQuery(payload);
+    if (!validateQuery(query, error))
+        return send_error(jsonErrorResponse(error, ErrorCodes::BAD_REQUEST_PARAMETER));
+
+    LOG_INFO(log, "Execute query: {}", query);
+    ReadBufferFromString in{query};
+
+    /// Prepare output buffer
+    const auto out = getOutputBuffer(response);
+    try
+    {
+        /// FIXME: implement outbound solution to support query progress
+        executeQuery(in, *out, /* allow_into_outfile = */ false, query_context, {});
+    }
+    catch (Exception & e)
+    {
+        tryLogCurrentException(log);
+        /// Send the error message into already used (and possibly compressed) stream.
+        /// Note that the error message will possibly be sent after some data.
+        /// Also HTTP code 200 could have already been sent.
+        http_status = HTTPResponse::HTTP_INTERNAL_SERVER_ERROR;
+        response.setStatusAndReason(HTTPResponse::HTTPStatus(http_status));
+        bool data_sent = out->count() != out->offset();
+
+        /// If buffer has data, and that data wasn't sent yet, then no need to send that data
+        if (!data_sent)
+            out->position() = out->buffer().begin();
+
+        writeString(jsonErrorResponse(e.message(), e.code()), *out);
+        writeChar('\n', *out);
+        out->next();
+        out->finalize();
+    }
+
+    return "";
+}
+
+String SearchHandler::getQuery(const Poco::JSON::Object::Ptr & payload) const
+{
+    /// Setup query settings
+    const auto & mode = payload->has("mode") ? payload->get("mode").toString() : "standard";
+
+    /// FIXME: to support 'realtime' mode in future
+    /// FIXME: change the behavior of 'verbose' mode. Currently in frontier service, 'standard' and 'verbose' mode are the same
```

Comment:
For `verbose` mode, enable these 2 flags

```
  query_context->setSetting("asterisk_include_materialized_columns", true);
  query_context->setSetting("asterisk_include_alias_columns", true);
```

Reply:
In frontier service, "standard" and "verbose" mode are the same and by default, "standard" mode enble these two flags. Should we change the behavior here?

Reply:
This is a `TODO` in frontier which is not implemented